hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-resourcemanager/src/main/java/org/apache/hadoop/yarn/server/resourcemanager/security/DelegationTokenRenewer.java

```diff
@@ -523,9 +528,14 @@ private void handleAppSubmitEvent(AbstractDelegationTokenRenewerAppEvent evt)
             tokenConf = getConfig();
           }
           dttr = new DelegationTokenToRenew(Arrays.asList(applicationId), token,
-              tokenConf, now, shouldCancelAtEnd, evt.getUser());
+              tokenConf, tokenExpiredTime.get(), shouldCancelAtEnd, evt.getUser());
+
           try {
-            renewToken(dttr);
+            // if expire date is not greater than now, renew token.
+            // add extra time in case of clock skew
+            if (tokenExpiredTime.get() <= now + clockSkewExtraTime) {
```

Comment:
Sorry I didn't get why add clock skew condition check here. I think we should file another JIRA to fix if it is issue indeed.